cmd/gctcli/commands.go

```diff
@@ -4723,17 +4723,309 @@ func findMissingSavedCandleIntervals(c *cli.Context) error {
+		},
+		&cli.StringFlag{
+			Name:    "pair",
+			Aliases: []string{"p"},
+			Usage:   "the currency pair",
+		},
+		&cli.StringFlag{
+			Name:        "start",
+			Aliases:     []string{"sd"},
+			Usage:       "<start> rounded down to the nearest hour, ensure your starting position is within this window for accurate calculations",
+			Value:       time.Now().AddDate(-1, 0, 0).Truncate(time.Hour).Format(common.SimpleTimeFormat),
+			Destination: &startTime,
+		},
+		&cli.StringFlag{
+			Name:        "end",
+			Aliases:     []string{"ed"},
+			Usage:       "<end> rounded down to the nearest hour, ensure your last position is within this window for accurate calculations",
+			Value:       time.Now().Truncate(time.Hour).Format(common.SimpleTimeFormat),
+			Destination: &endTime,
+		},
+		&cli.IntFlag{
+			Name:        "limit",
+			Aliases:     []string{"l"},
+			Usage:       "the number of positions (not orders) to return",
+			Value:       86400,
+			Destination: &limit,
+		},
+		&cli.StringFlag{
+			Name:    "status",
+			Aliases: []string{"s"},
+			Usage:   "limit return to position statuses - open, closed, any",
+			Value:   "ANY",
+		},
+		&cli.BoolFlag{
+			Name:    "verbose",
+			Aliases: []string{"v"},
+			Usage:   "includes all orders that make up a position in the response",
+		},
+		&cli.BoolFlag{
+			Name:    "overwrite",
+			Aliases: []string{"o"},
+			Usage:   "if true, will overwrite futures results for the provided exchange, asset, pair",
+		},
+	},
+}
+
+func getFuturesPositions(c *cli.Context) error {
+	if c.NArg() == 0 && c.NumFlags() == 0 {
+		return cli.ShowCommandHelp(c, "getfuturesposition")
+	}
+
+	var exchangeName string
+	if c.IsSet("exchange") {
+		exchangeName = c.String("exchange")
+	} else {
+		exchangeName = c.Args().First()
+	}
+
+	var assetType string
+	if c.IsSet("asset") {
+		assetType = c.String("asset")
+	} else {
+		assetType = c.Args().Get(1)
+	}
+
+	if !validAsset(assetType) {
+		return errInvalidAsset
+	}
+
+	var currencyPair string
+	if c.IsSet("pair") {
+		currencyPair = c.String("pair")
+	} else {
+		currencyPair = c.Args().Get(2)
+	}
+	if !validPair(currencyPair) {
+		return errInvalidPair
+	}
+
+	p, err := currency.NewPairDelimiter(currencyPair, pairDelimiter)
+	if err != nil {
+		return err
+	}
+
+	if !c.IsSet("start") {
+		if c.Args().Get(3) != "" {
+			startTime = c.Args().Get(3)
+		}
+	}
+
+	if !c.IsSet("end") {
+		if c.Args().Get(4) != "" {
+			endTime = c.Args().Get(4)
+		}
+	}
+	if c.IsSet("limit") {
+		limit = c.Int("limit")
+	} else if c.Args().Get(5) != "" {
+		var limit64 int64
+		limit64, err = strconv.ParseInt(c.Args().Get(5), 10, 64)
+		if err != nil {
+			return err
+		}
+		limit = int(limit64)
+	}
+
+	var status string
+	if c.IsSet("status") {
+		status = c.String("status")
+	} else if c.Args().Get(6) != "" {
+		status = c.Args().Get(6)
+	}
+	if !strings.EqualFold(status, "any") &&
+		!strings.EqualFold(status, "open") &&
+		!strings.EqualFold(status, "closed") &&
+		status != "" {
```

Comment:
Why?

Reply:
I guess if it was unset by accident?

Reply:
The check is there to prevent people from seeing the help description of ` "limit return to position statuses - open, closed, any"`, entering `lol` and expecting a response

Reply:
Ahh okay, I get you. I just assumed an empty string was invalid. 👍

Reply:
Oh, with the way github renders, I thought you were asking to remove the entire `if` statement, not just the `""` check. I went with an empty string assuming "any" and made it valid.

It also doesn't look like I do any filtering on the RPCServer side, so I'll add that